### PR TITLE
Simplify provider interfaces to set error messages

### PR DESCRIPTION
**Pull request by jessesuen**
* Rename AnalysisProvider to MetricProvider
* Add message field to measurements
* Simplify provider interface so that it never returns error, and error messages are expected to be set in the measurement.

### CODE REVIEW

File: providers/provider.go [hunk 1]
```diff
@@ -16,11 +16,11 @@ import (
 type Provider interface {
```
Comment (dthomson25) [inline providers/provider.go]: To your point earlier, do we want to rerun the package too?

**Comment by dthomson25** [comment]
Do we not have a unit test for executing the resume provider?

https://codecov.io/gh/argoproj/argo-rollouts/pull/189/diff#D11-194

File: providers/provider.go (hunk 1 above)
Comment (jessesuen) [reply providers/provider.go]: Sure. I will rename to metric_providers

File: analysis/analysis.go [hunk 2]
```diff
@@ -213,18 +213,14 @@ func (c *AnalysisController) runMeasurements(run *v1alpha1.AnalysisRun, tasks []
 					metricResult.Count++
 				case v1alpha1.AnalysisStatusError:
 					metricResult.Error++
+					log.Warnf("metric errored: %s", newMeasurement.Message)
```
[the remark this answers is not in the document]
Comment (jessesuen) [reply analysis/analysis.go]: Sure.

**Pull request merged**